scripts/dts/python-devicetree/src/devicetree/edtlib.py

```diff
@@ -90,6 +91,12 @@
 from devicetree.grutils import Graph
 from devicetree._private import _slice_helper
 
+def _compute_hash(path: str) -> str:
+    # Calculates the hash associated with the node's full path.
+    hasher = hashlib.sha256()
+    hasher.update(path.encode())
+    return hasher.hexdigest()
+
```

Comment:
A few thoughts:
* we could save space (~33%) by base64-encoding the hash's raw bytes
  * Limitation: the resulting hash must be a valid C identifier
    * base64 uses `+`, `/` and `=` which are illegal...
    * base32 saves only ~20% but doesn't have this problem
* appending a known-unique value (e.g., node address) to the hash string would guarantee that collisions can *never* happen
  * not sure this is worth the trouble with how unlikely SHA-256 collisions are, though

Reply:
Thanks for the idea!

> base64 uses `+`, `/` and `=` which are illegal...

True, but we don't really need an invertible transformation, just a hash, so we can cheat a little:  
 * `=` is only used in padding, so we can drop it. 
 * Replacing both `+` and `/` with the legal symbol `_` would make clashes _a little_ more probable, but still vanishingly small: ChatGPT said with a convincing formula that it's 1.23 times more probable than a direct SHA256 collision (5.5E-78...).

At the same time, though, the identifier would become "only" 43 chars. Looks like a nice tradeoff. :tada: